bench/vtile-encode.cpp

```diff
@@ -0,0 +1,169 @@
+#include <mapnik/timer.hpp>
+#include <mapnik/util/file_io.hpp>
+#include "vector_tile_backend_pbf.hpp"
+#include "vector_tile_processor.hpp"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#pragma GCC diagnostic ignored "-Wsign-conversion"
+#include "vector_tile.pb.h"
+#pragma GCC diagnostic pop
+
+#include <iostream>
+#include <fstream>
+
+// mapnik
+#include <mapnik/global.hpp>
+#include <mapnik/box2d.hpp>
+#include <mapnik/well_known_srs.hpp>
+#include <mapnik/json/geometry_parser.hpp>
+#include <mapnik/memory_datasource.hpp>
+#include <mapnik/feature_factory.hpp>
+#include <mapnik/map.hpp>
+#include <mapnik/layer.hpp>
+#include <mapnik/request.hpp>
+
+#include <protozero/pbf_reader.hpp>
+
+typedef mapnik::vector_tile_impl::backend_pbf backend_type;
+typedef mapnik::vector_tile_impl::processor<backend_type> renderer_type;
+
+void from_pixels(double shift, double & x, double & y)
+{
+    double b = shift/2.0;
+    x = (x - b)/(shift/360.0);
+    double g = (y - b)/-(shift/(2 * M_PI));
+    y = mapnik::R2D * (2.0 * std::atan(std::exp(g)) - mapnik::M_PI_by2);
+}
+
+void xyz(int tile_size,
+    int x,
+    int y,
+    int z,
+    double & minx,
+    double & miny,
+    double & maxx,
+    double & maxy)
+{
+    minx = x * tile_size;
+    miny = (y + 1.0) * tile_size;
+    maxx = (x + 1.0) * tile_size;
+    maxy = y * tile_size;
+    double shift = std::pow(2.0,z) * tile_size;
+    from_pixels(shift,minx,miny);
+    from_pixels(shift,maxx,maxy);
+    mapnik::lonlat2merc(&minx,&miny,1);
+    mapnik::lonlat2merc(&maxx,&maxy,1);
+}
+
+
+int main(int argc, char** argv)
+{
+    try
+    {
+
+        if (argc < 4)
+        {
+            std::clog << "usage: vtile-encode /path/to/geometry.geojson z x y [iterations]\n";
+            return -1;
+        }
+        std::string geojson(argv[1]);
+        mapnik::util::file input(geojson);
+        if (!input.open())
+        {
+            std::clog << std::string("failed to open ") + geojson << "\n";
+            return -1;
+        }
+
+        int z = std::stoi(argv[2]);
+        int x = std::stoi(argv[3]);
+        int y = std::stoi(argv[4]);
+
+        std::size_t iterations = 100;
+        std::string output_path = "";
+        if (argc > 5)
+        {
+            for (int i = 5; i < argc; i++)
+            {
+                std::string flag = argv[i];
+                if (flag == "-i")
+                {
+                    iterations = std::stoi(argv[i + 1]);
+                }
+                if (flag == "-o")
+                {
+                    output_path = argv[i + 1];
+                }
+            }
+        }
+
+        std::clog << "z:" << z << " x:" << x << " y:" << y <<  " iterations:" << iterations << "\n";
```

Comment:
Instead of "\n" you can use `std::endl` which will work for windows as well, not a big deal just a FYI. @jakepruitt